File: HashTable/Happy Number/Happy_Number.java
public class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> set = new HashSet<>();
        while(!set.contains(n)) {
            set.add(n);
            int result = 0;
            while ( n != 0 ) {
                result += (n%10)*(n%10);
                n /= 10;
            }
            if (result == 1){
                return true;
            } else  {
                n = result;
            }
        }
        return false;
    }
}

File: HashTable/Happy Number/Happy_Number.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        dict_buff = {}
        while True:
            dict_buff[n] = True;
            n = sum(int(x)*int(x) for x in list(str(n)));
            if n == 1 or n in dict_buff:
                break;
        return n == 1;